x-pack/metricbeat/module/azure/billing/service.go

```diff
@@ -6,61 +6,88 @@ package billing
 
 import (
 	"context"
+	"fmt"
 	"time"
 
-	"github.com/Azure/go-autorest/autorest/date"
-
-	"github.com/Azure/azure-sdk-for-go/services/consumption/mgmt/2019-10-01/consumption"
-	//"github.com/Azure/azure-sdk-for-go/services/costmanagement/mgmt/2019-11-01/costmanagement"
-	"github.com/Azure/go-autorest/autorest/azure/auth"
-
 	"github.com/elastic/beats/v7/x-pack/metricbeat/module/azure"
 	"github.com/elastic/elastic-agent-libs/logp"
 
-	"github.com/Azure/azure-sdk-for-go/services/costmanagement/mgmt/2019-11-01/costmanagement"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/consumption/armconsumption"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/costmanagement/armcostmanagement"
 )
 
 // Service offers access to Azure Usage Details and Forecast data.
 type Service interface {
-	GetForecast(scope string, startTime, endTime time.Time) (costmanagement.QueryResult, error)
+	GetForecast(
+		scope string,
+		startTime,
+		endTime time.Time,
+	) (armcostmanagement.QueryResult, error)
 	GetUsageDetails(
 		scope string,
 		expand string,
 		filter string,
 		skipToken string,
 		top *int32,
-		metricType consumption.Metrictype,
+		metricType armconsumption.Metrictype,
 		startDate string,
-		endDate string) (consumption.UsageDetailsListResultPage, error)
+		endDate string,
+	) (armconsumption.UsageDetailsListResult, error)
 }
 
 // UsageService is a thin wrapper to the Usage Details API and the Forecast API from the Azure SDK for Go.
 type UsageService struct {
-	usageDetailsClient *consumption.UsageDetailsClient
-	forecastClient     *costmanagement.ForecastClient
+	usageDetailsClient *armconsumption.UsageDetailsClient
+	forecastClient     *armcostmanagement.ForecastClient
 	context            context.Context
 	log                *logp.Logger
 }
 
 // NewService builds a new UsageService using the given config.
 func NewService(config azure.Config) (*UsageService, error) {
-	clientConfig := auth.NewClientCredentialsConfig(config.ClientId, config.ClientSecret, config.TenantId)
-	clientConfig.AADEndpoint = config.ActiveDirectoryEndpoint
-	clientConfig.Resource = config.ResourceManagerEndpoint
-	authorizer, err := clientConfig.Authorizer()
+	credential, err := azidentity.NewClientSecretCredential(config.TenantId, config.ClientId, config.ClientSecret, &azidentity.ClientSecretCredentialOptions{
+		ClientOptions: policy.ClientOptions{
+			Cloud: cloud.Configuration{
+				Services:                     cloud.AzurePublic.Services,
+				ActiveDirectoryAuthorityHost: config.ActiveDirectoryEndpoint,
+			},
+		},
+	})
```

Comment:
Did you find a way to pass the "Active Directory Endpoint" and "Resource Manager Endpoint" from the options?